src/targets/gpu/mlir.cpp

```diff
@@ -503,6 +508,17 @@ struct mlir_program
         {
         }
 
+        void set_operand_segment_sizes(int num_segments, const std::vector<int>& sizes)
+        {
+            MlirAttribute segment_sizes_attr =
+                mlirDenseI32ArrayGet(prog->ctx.get(), num_segments, sizes.data());
```

Comment:
From here, it looks like `num_segments` is the size of the vector. Why not just do `sizes.size()` instead of `num_segments`?